### PR TITLE
fix(NcCheckboxRadioSwitch): Pass attrs to `input` if available

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

I would expect the attrs on the input element to allow any aria attributes.
So if not button type, pass the attrs to the input element to allow setting aria tags like `aria-invalid` and `aria-errormessage`

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

**Review by raimund-schluessler — APPROVED** [review]

**Review by ShGKme — APPROVED** [review]
Why don't pass to the bottom as well?

**Comment by susnux** [comment]
> Why don't pass to the button as well?

They are as in this case it is the root element.

**Comment by susnux** [comment]
we maybe should add consistent props for all input elements to have some focus and validation API independent of the implementation.

**Comment by ShGKme** [comment]
> They are as in this case it is the root element.

But it is disabled via
`inheritAttrs: false,`

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue [hunk 1]
```diff
@@ -322,6 +326,9 @@ export default {
 		NcCheckboxContent,
 	},
 
+	// We need to pass attributes to the input element
+	inheritAttrs: false,
```
Comment (ShGKme) [inline src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: > They are as in this case it is the root element.

But it is disabled via `inheritAttrs: false` here

Comment (susnux) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: For vue2 class and style are passed to the root as they are not attrs but somewhat special

Comment (susnux) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: https://v3-migration.vuejs.org/breaking-changes/attrs-includes-class-style#_2-x-behavior

Comment (ShGKme) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: > Why don't pass to the button as well?

Here I meant the attrs, not only class/style.

Comment (ShGKme) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: Currently there is a new possibility to pass attrs to `input`, but the possibility to pass attrs to the `button` has been disabled.

Comment (susnux) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: No for button type they are passed:
https://github.com/nextcloud-libraries/nextcloud-vue/pull/5507/files#diff-80b068a320cd09d28f4f8913f4c77c73754bed31f69a9b3a25353059d8573623R279

Comment (susnux) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:330]: So class & style are always set to the wrapper.
`attrs` are passed to the wrapper (the button) for button type and to the input for other types.

**Comment by skjnldsv** [comment]
Breaks server

**Comment by susnux** [comment]
> we can't apply attributes to the outer span now, that is an issue imho

The structure of components is private and can change anytime, thus assumptions about it will always be prone to break.
That is why we have e.g. `content-classes` etc for assigning custom classes to dialogs without any assumption about internal class names or structure.
In this case you have a checkbox thus you would expect to have a checkbox element (input).

This "regression" is only about tests that assume some specific internal structure, but it is more a fix.

If you currently have `cy.get('[data-cy-my-checkbox] input')` you could simplify to  `cy.get('[data-cy-my-checkbox]')` or the most secure way: `cy.findByRole('checkbox', { name: 'My checkbox' })` which will work regardless of the internal structure and only depends on your app code (the accessible name that you assign).

**Pull request merged**